core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -152,6 +152,11 @@ public static OpType fromString(String sOpType) {
 
     private String pipeline;
 
+    private long autoGeneratedTimestamp = -1;
```

Comment:
nit: can we make -1 a constant and doc it? It has proven useful in the past to be able to see usages of these constants.